extension/js/common/platform/store/contact-store.ts

```diff
@@ -14,14 +13,37 @@ import { OpenPGPKey } from '../../core/crypto/pgp/openpgp-key.js';
 type DbContactObjArg = {
   email: string,
   name?: string | null,
-  client?: 'pgp' | 'cryptup' | PgpClient | null,
   pubkey?: string | null,
-  pendingLookup?: boolean | number | null,
   lastUse?: number | null, // when was this contact last used to send an email
-  lastSig?: number | null, // last pubkey signature (when was pubkey last updated by owner)
   lastCheck?: number | null; // when was the local copy of the pubkey last updated (or checked against Attester)
 };
 
+type Email = {
+  email: string;
+  name: string | null;
+  searchable: string[];
+  fingerprints: string[];
+  lastUse: number | null;
+};
+
+type Pubkey = {
+  fingerprint: string;
+  armoredKey: string;
+  longids: string[];
+  lastCheck: number | null,
+  expiresOn: number | null;
+};
+
+type PubkeyAttributes = {
+  fingerprint: string | null;
+  expiresOn: number | null;
+};
+
+export type ContactV4 = {
```

Comment:
@tomholub this is a type I created to conveniently extract the contact info (from `emails` table) and all associated pubkeys. So far it is used internally only, but later we can use it from outside the ContactStore.